lib/ecto/adapters/postgres/connection.ex

```diff
@@ -702,6 +719,19 @@ if Code.ensure_loaded?(Postgrex.Connection) do
     defp null_expr(true), do: "NULL"
     defp null_expr(_), do: []
 
+    defp new_constraint_expr(%Ecto.Migration.Constraint{check: nil, exclude: nil})  do
```

Comment:
I defined matching function heads for the two bad cases so that I could give a friendlier message than "no matching function", but I'll delete these if you think that's overdoing it. :)